source/slang/check.cpp

```diff
@@ -1588,6 +1588,44 @@ namespace Slang
             return true;
         }
 
+        bool doesGenericSignatureMatchRequirement(
+            GenericDecl * genDecl,
+            DeclRef<GenericDecl> requirementGenDecl)
+        {
+            // TODO: genDecl should be a DeclRef to capture the environment generic variables needed to get 
+            // a concrete type for a generic constraint super type (e.g. when this member belongs to a generic type)
+
+            if (genDecl->Members.Count() != requirementGenDecl.getDecl()->Members.Count())
+                return false;
+            for (UInt i = 0; i < genDecl->Members.Count(); i++)
+            {
+                auto genMbr = genDecl->Members[i];
+                auto requiredGenMbr = genDecl->Members[i];
+                if (auto genTypeMbr = genMbr.As<GenericTypeParamDecl>())
+                {
+                    if (auto requiredGenTypeMbr = requiredGenMbr.As<GenericTypeParamDecl>())
+                    {
+                    }
+                    else
+                        return false;
+                }
```

Comment:
It isn’t critical to handle right now, but there should also be a case here for `GenericValueParamDecl`.